translator/util/sdkutil.go

```diff
@@ -93,15 +87,13 @@ func detectRegion(mode string, credsConfig map[string]string) (region string) {
 
 	// For ec2, fallback to metadata when no region info found in credential profile.
 	if region == "" && mode == config.ModeEC2 {
-		region = defaultEC2Region()
-	}
-
-	// try to get region from ecs metadata
-	if region == "" && mode == config.ModeEC2 {
-		fmt.Println("I! detect region from ecs")
-		region = defaultECSRegion()
+		fmt.Println("I! Detect region from EC2/ECS")
+		if region = defaultEC2Region(); region == "" {
+			// Try to get region from ecs metadata
+			region = defaultECSRegion()
+		}
 	}
```

Comment:
This _looks_ equivalent to me, but is there a way to unit test this?

Reply:
For ECS, an integration test is needed since I have not known to replicate the `Metadataendpont`. For a EC2, it would be the same since it also depends on [EC2Metadata endpoint](https://github.com/aws/amazon-cloudwatch-agent/blob/master/tool/util/util.go#L136-L155)

Reply:
On second though, take back my word a little bit. It seems there are a workaround in [build an AWSVPC Fargate](https://github.com/aws/amazon-cloudwatch-agent/blob/master/internal/ecsservicediscovery/decoratedtask_test.go#L50-L136). Would need to take a look at it.